native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/SyntaxTreeGenerator.java

```diff
@@ -131,7 +132,9 @@ public static SyntaxTree generateSyntaxTree(StubFile stubFile, boolean isRoot) {
             if (checkForImportsInServices(methodList, "time:Utc")
                     || checkForImportsInServices(methodList, "time:Seconds")) {
                 importTime = true;
-                break;
+            }
+            if (checkForImportsInServices(methodList, "'any:Any")) {
+                importAny = true;
```

Comment:
We can generate the imports by checking the imports in the proto definition. But there can be proto files where the import is not used. eg:
```
syntax = "proto3";

import "google/protobuf/wrappers.proto";
import "google/protobuf/timestamp.proto";

service helloWorld {
	rpc serverCall (google.protobuf.Timestamp) returns (stream HelloResponse);
}

message HelloResponse {
	string message = 1;
}
```

Here the `wrappers` import is never used. Therefore, in the current implementation, we search for the usage in the services list and message map and generate necessary imports. This is to avoid compilation errors in the generated stub file due to unused imports.

A possible improvement is to give an error if a certain import is not used in the proto definition. But then again we need to go through the services list and message map to find the unused imports.

Reply:
In golang, they are giving an error when there is an unused import. I think this is already supported by the protoc compiler

```
warning: Import google/protobuf/wrappers.proto is unused.
```

Reply:
Added a warning message and updated the imports generation logic in [813659b](https://github.com/ballerina-platform/module-ballerina-grpc/pull/739/commits/813659b6f4d9d01491fac9e38f61c9b2d1e7cde5).